download_and_create_reference_datasets/v03/write_combined_interval_reference_ht.py

```diff
@@ -49,16 +24,26 @@ def run(environment: str, dataset: str):
         genome_version=genome_version,
     )
     if hl.hadoop_exists(os.path.join(destination_path, '_SUCCESS')):
-        ht = update_existing(destination_path, dataset, genome_version)
+        ht = update_existing_joined_hts(
+            destination_path,
+            dataset,
+            DATASETS,
+            VERSION,
+            genome_version,
+        )
     else:
-        ht = create_new(genome_version)
+        ht = join_hts(DATASETS, VERSION, reference_genome=genome_version)
     print(f'Uploading ht to {destination_path}')
     write_ht(ht, destination_path)
```

Comment:
I don't think this works since you're reading and writing to the same path. Adding a checkpoint to some scratch storage location in the update existing function is probably the easiest way to get this working.

Reply:
hrm... I think the combined interval reference doesn't have a version in the path name, but the others do.  Given that we now have individual dataset versions I think we don't need to keep the version in the pathname and we should just be updating a single file?  so we don't have to store a separate copy every time we update?

Reply:
Which indicates both pathways should checkpoint.